lib/migrate.js

```diff
@@ -0,0 +1,448 @@
+          },
+          function addIndexes(cb) {
+            createIndexes(results.indexesToAdd, cb);
+          },
+        ], cb);
+      });
+
+      function createIndexes(indexes, cb) {
+        assert(typeof indexes === 'object', 'indexes to create must be an object!');
+        async.eachOf(indexes, create, cb);
+        // {indexName: "foo"} or {indexName: [{"foo": "asc"}, {"bar": "asc"}]}
+        function create(value, key, cb) {
+          createIndex(key, value, cb);
+        };
+      };
+
+      function createIndex(name, fields, cb) {
+      // naming convertion: '_design/LBModel__Foo__LBIndex__foo_index',
+      // here the driver api takes in the name without prefix '_design/'
+        var config = {
+          ddocName: self.getIndexModelPrefix(mo) + '__' + model + '__' +
+          self.getIndexPropertyPrefix(mo) + '__' + name,
+          indexName: name,
+          fields: fields,
+        };
+        self.createIndex(config.ddocName, config.indexName, config.fields, cb);
+      }
+
+      function removeIndexes(indexes, cb) {
+        assert(typeof indexes === 'object', 'indexes to drop must be an object!');
+        async.eachOf(indexes, removeIndex, cb);
+        // {ddoc: "_design/ddocName", name: "indexName"}
+        function removeIndex(value, key, cb) {
+          self.deleteIndex(mo, value.ddoc, cb);
+        }
+      };
+    };
+  };
+
+  /**
+ * Create an index in couchdb, you can specify the ddocName and indexName
+ * @param {String} ddocName design doc name with prefix '_design/'
+ * @param {String} indexName index name
+ * @param {Array} fields example format: [{field1: 'asc'}, {field2: 'asc'}]
+ * @callback {Function} cb The callback function
+ */
+  CouchDB.prototype.createIndex = function(ddocName, indexName, fields, cb) {
+    debug('createIndex: ddocName %s, indexName %s, fields %s', ddocName,
+      indexName, fields);
+    assert(Array.isArray(fields), 'fields in the index must be an array!');
+    assert(typeof ddocName === 'string', 'ddocName of index must be an array!');
+    assert(typeof indexName === 'string', 'indexName in the index must be an array!');
+
+    var self = this;
+    var indexBody = {
+      index: {
+        fields: fields,
+      },
+      ddoc: ddocName,
+      name: indexName,
+      type: 'json',
+    };
+
+    var requestObject = {
+      db: self.settings.database,
+      path: '_index',
+      method: 'post',
+      body: indexBody,
+    };
+
+    self.couchdb.request(requestObject, cb);
+  };
+
+  /**
+ * Return modify index results with: `indexesToDrop`, `indexesToAdd`
+ * @param {Object} mo the model object returned by this.selectModel(modelName)
+ * @param {Object} newLBIndexes returned from juggler's modelDefinition.indexes()
+ * @param {Boolean} isMigrate flag to tell do we want to compare new/old indexes
+ * @callback {Function} cb The callback function
+ * @param {Object} result indexes to modify in the following format:
```

Comment:
I think `@returns` is for sync function's return properties.
for async function, the annotation is the one I use:
search 'are optional if the function' on page https://github.com/strongloop/strong-docs